sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -0,0 +1,1501 @@
+            namespace=namespace,
+            job_kind=constants.TFJOB_KIND,
+            job_plural=constants.TFJOB_PLURAL,
+            delete_options=delete_options,
+        )
+
+    # ------------------------------------------------------------------------ #
+    # PyTorchJob Training Client APIs.
+    # ------------------------------------------------------------------------ #
+    def create_pytorchjob(
+        self,
+        pytorchjob: models.KubeflowOrgV1PyTorchJob,
+        namespace=utils.get_default_target_namespace(),
+    ):
+        """Create the PyTorchJob.
+
+        Args:
+            pytorchjob: PyTorchJob object of type KubeflowOrgV1PyTorchJob.
+            namespace: Namespace for the PyTorchJob.
+
+        Raises:
+            TimeoutError: Timeout to create PyTorchJob.
+            RuntimeError: Failed to create PyTorchJob.
+        """
+
+        utils.create_job(
+            custom_api=self.custom_api,
+            job=pytorchjob,
+            namespace=namespace,
+            job_kind=constants.PYTORCHJOB_KIND,
+            job_plural=constants.PYTORCHJOB_PLURAL,
+        )
+
+    def create_pytorchjob_from_func(
+        self,
+        name: str,
+        func: Callable,
+        parameters: Dict[str, Any] = None,
+        base_image: str = constants.PYTORCHJOB_BASE_IMAGE,
+        namespace: str = utils.get_default_target_namespace(),
+        num_worker_replicas: int = None,
+        packages_to_install: List[str] = None,
+        pip_index_url: str = "https://pypi.org/simple",
+    ):
+        """Create PyTorchJob from the function.
+
+        Args:
+            name: Name for the PyTorchJob.
+            func: Function that PyTorchJob uses to train the model. This function
+                must be Callable. Optionally, this function might have one dict
+                argument to define input parameters for the function.
+            parameters: Dict of input parameters that training function might receive.
+            base_image: Image to use when executing the training function.
+            namespace: Namespace for the PyTorchJob.
+            num_worker_replicas: Number of Worker replicas for the PyTorchJob.
+                If number of Worker replicas is 1, PyTorchJob uses only
+                Master replica.
+            packages_to_install: List of Python packages to install in addition
+                to the base image packages. These packages are installed before
+                executing the objective function.
+            pip_index_url: The PyPI url from which to install Python packages.
+        """
+
+        # Check if at least one worker replica is set.
+        if num_worker_replicas is None:
+            raise ValueError("At least one Worker replica for PyTorchJob must be set")
+
+        # Check if function is callable.
+        if not callable(func):
+            raise ValueError(
+                f"Training function must be callable, got function type: {type(func)}"
+            )
```

Comment:
Maybe, we can remove those validations once we introduce CEL validations.

Ref: https://github.com/kubeflow/training-operator/issues/1708

Reply:
Not sure, if we can validate training function in the CEL validation tho.

Reply:
Sorry for the confusion.
I meant only L870~L872. Maybe, we can validate `num_worker_replicas` in the following:

```shell
+kubebuilder:validation:XValidation:rule="self['Master'].replicas == 0 && self['Worker'].replicas == 0"
```